doc/nrf/device_guides/working_with_nrf/nrf91/nrf9160.rst

```diff
@@ -182,22 +182,163 @@ In Zephyr, the firmware for the application core of :ref:`zephyr:nrf9160dk_nrf91
+
+   .. figure:: images/programmer_selectdevice_nrf9160.png
+      :alt: Programmer - Select device
+
+      Programmer - Select device
+
+   If the DK is not visible, press ``Ctrl+R`` in Windows (``command+R`` in macOS) to restart the Programmer application.
+
+   The drop-down text changes to the type of the selected device, with its SEGGER ID below the name.
+   The **Device memory layout** section also changes its name to the device name, and indicates that the device is connected.
+   If the :guilabel:`Auto read memory` option is selected in the **DEVICE** section, the memory layout will update.
+   If it is not selected and you wish to see the memory layout, click :guilabel:`Read` in the **DEVICE** section.
+
+#. Click :guilabel:`Add file` in the **FILE** section, and select :guilabel:`Browse`.
+
+   .. figure:: images/programmer_addfile_nrf9160dk.png
+      :alt: Programmer - Add file
+
+      Programmer - Add file
+
+#. Navigate to where you extracted the firmware, and then to the :file:`img_app_bl` folder there.
+#. Select the :file:`.hex` file for the application you are programming.
+
+   For Asset Tracker v2, select the :file:`nrf9160dk_asset_tracker_v2_<version-number>.hex` HEX file.
+
+   For NB-IoT, there is a second variant of the Asset Tracker v2 firmware in the :file:`nrf9160dk_asset_tracker_v2_nbiot_legacy_pco_<version-number>.hex` file.
+   Only use this legacy variant if your network does not support ePCO.
+
+#. Click the :guilabel:`Erase & write` button in the **DEVICE** section to program the DK.
+   Do not unplug or turn off the DK during this process.
+
+   .. figure:: images/programmer_erasewrite_nrf9160dk.png
+      :alt: Programmer - Erase & write
+
+      Programmer - Erase & write
+
 .. _build_pgm_nrf9160:
 
 Building and programming
 ************************
 
 You can program applications and samples on the nRF9160 DK after obtaining the corresponding firmware images.
 
-Download the latest application and modem firmware from the `nRF9160 DK Downloads`_ page.
-
-To program applications using the Programmer app from `nRF Connect for Desktop`_, follow the instructions in :ref:`nrf9160_gs_updating_fw_application`.
-In Step 2, set the switch to **nRF91** or **nRF52** as appropriate for the application or sample you are programming.
-See the `Device programming section in the nRF9160 DK User Guide`_ for more information.
-Likewise, in Step 7, choose the :file:`.hex` file for the application you are programming.
-
 .. note::
-   When you update the application firmware on an nRF9160 DK, you must also update the modem firmware as described in :ref:`nrf9160_gs_updating_fw_modem`.
+   When you update the application firmware on an nRF9160 DK, make sure the modem firmware and application firmware are compatible versions.
+
+To program applications using the Programmer app from `nRF Connect for Desktop`_, follow the instrutions in :ref:`nrf9160_updating_fw_application`.
```

Comment:
May be we could remove this sentence because the section is just above this section.